### PR TITLE
Remove delete for event type

### LINKED ISSUE #4992: SoftDeletion for event-types doesn't make sense
**Describe the bug**

SoftDeletion on event types don't make sense since they're always there, and they won't be modified.

Their schemas shouldn't inherit from SoftDeletion like other event or user properties.

### PR DESCRIPTION
**Pull request by Kreijstal**
since it doesn't need to be soft deleted, or deleted for that reason

Fixes #4992 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Event types shouldn't be solft deletable

### CODE REVIEW

**Comment by srv-twry** [comment]
Why won't they be deleted ? The admin can always remove them if he/she wants. We provide a starting set of event-types which can be easily modified by the admin according to his/her needs.

File: app/api/event_types.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def before_get_object(self, view_kwargs):
             else:
                 view_kwargs['id'] = None
 
-    decorators = (api.has_permission('is_admin', methods="PATCH,DELETE"),)
+    decorators = (api.has_permission('is_admin', methods="PATCH"),)
```
Comment (srv-twry) [inline app/api/event_types.py]: Removing `DELETE` from here doesn't mean that this endpoint won't support deletion. Removing it from decorators means that we won't check for admin access when deleting them which is even worse. Just for information although I don't think this PR is needed.

**Comment by mayank8318** [comment]
@Kreijstal I agree with @srv-twry here. The admin can edit/delete/add event-types as per requirements

**Comment by srv-twry** [comment]
@Kreijstal Please close this now.

**Comment by srv-twry** [comment]
Closing this one. Feel free to reopen if you don't agree with any of the comments above.

**Pull request closed without merging**